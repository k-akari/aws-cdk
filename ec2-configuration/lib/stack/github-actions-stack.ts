import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { IamRole } from '../resource/iam-role';

export class GithubActionsStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    new IamRole().createGithubRole(this);
  }
}
