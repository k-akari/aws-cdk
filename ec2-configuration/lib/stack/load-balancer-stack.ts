import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc, CfnSubnet, CfnSecurityGroup } from 'aws-cdk-lib/aws-ec2';
import { CfnInstance } from 'aws-cdk-lib/aws-ec2';
import { Elb } from '../resource/elb';

export class LoadBalancerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: CfnSubnet, sg: CfnSecurityGroup, ec2Instance: CfnInstance, props?: StackProps) {
    super(scope, id, props);
   
    new Elb(this, vpc, subnet, sg, ec2Instance);
  }
}
