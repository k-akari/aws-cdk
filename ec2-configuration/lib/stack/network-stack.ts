import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { VPC } from '../resource/vpc';
import { Subnet } from '../resource/subnet';
import { InternetGateway } from '../resource/internet-gateway';
import { RouteTable } from '../resource/route-table';
import { NetworkAcl } from '../resource/network-acl';

export class NetworkStack extends Stack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new VPC(this);
    this.vpc = vpc.vpc;
    this.subnet = new Subnet(this, this.vpc);
    const internetGateway = new InternetGateway(this, this.vpc);
    new RouteTable(this, this.vpc, this.subnet, internetGateway.igw);
    new NetworkAcl(this, this.vpc, this.subnet);
  }
}
