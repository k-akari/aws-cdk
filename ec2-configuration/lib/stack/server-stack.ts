import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { Ec2 } from '../resource/ec2';
import { Subnet } from '../resource/subnet';
import { SecurityGroup } from '../resource/security-group';
import { IamRole } from '../resource/iam-role';
import { ElasticIp } from '../resource/elastic-ip';
import { Elb } from '../resource/elb';

export class ServerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: Subnet, props?: StackProps) {
    super(scope, id, props);

    const sg = new SecurityGroup(this, vpc);

    const instanceProfile = new IamRole().createEc2InstanceProfile(this);
    const ec2Instance = new Ec2().createEc2Instance(this, subnet.public1a, instanceProfile, sg.ec2);
    new ElasticIp().associateElasticIp(this, ec2Instance, 'ec2-1a');

    new Elb(this, vpc, [subnet.public1a, subnet.public1c], sg.alb, ec2Instance);
  }
}
