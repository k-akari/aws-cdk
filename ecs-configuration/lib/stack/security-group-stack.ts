import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { SecurityGroup } from '../resource/security-group';

export class SecurityGroupStack extends Stack {
  public readonly sg: SecurityGroup;

  constructor(scope: Construct, id: string, vpc: Vpc, props?: StackProps) {
    super(scope, id, props);

    this.sg = new SecurityGroup(this, vpc);
  }
}
