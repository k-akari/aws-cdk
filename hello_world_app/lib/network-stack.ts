import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from './resource/vpc';
import { Subnet } from './resource/subnet';
import { InternetGateway } from './resource/internetGateway';
import { ElasticIp } from './resource/elasticIp';
import { NatGateway } from './resource/natGateway';
import { RouteTable } from './resource/routeTable';
import { NetworkAcl } from './resource/networkAcl';

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    // VPC
    const vpc = new Vpc();
    vpc.createResources(this);

    // Subnet
    const subnet = new Subnet(vpc.vpc);
    subnet.createResources(this);

    // Internet Gateway
    const internetGateway = new InternetGateway(vpc.vpc);
    internetGateway.createResources(this);

    // Elastic IP
    const elasticIp = new ElasticIp();
    elasticIp.createResources(this);

    // NAT Gateway
    const natGateway = new NatGateway(
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      elasticIp.ngw1a,
      elasticIp.ngw1c
    );
    natGateway.createResources(this);

    // Route Table
    const routeTable = new RouteTable(
      vpc.vpc,
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      subnet.subnetPrivate1a,
      subnet.subnetPrivate1c,
      internetGateway.igw,
      natGateway.ngw1a,
      natGateway.ngw1c
    );
    routeTable.createResources(this);

    // Network ACL
    const networkAcl = new NetworkAcl(
      vpc.vpc,
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      subnet.subnetPrivate1a,
      subnet.subnetPrivate1c
    );
    networkAcl.createResources(this);
  }
}
