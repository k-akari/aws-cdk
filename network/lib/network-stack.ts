import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from "aws-cdk-lib/aws-ec2";

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const vpc = new Vpc(this, "sample-vpc", {});
  }
}
