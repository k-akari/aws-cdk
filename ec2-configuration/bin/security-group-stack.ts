#!/usr/bin/env node
import 'source-map-support/register';
import * as cdk from 'aws-cdk-lib';
import { NetworkStack } from '../lib/stack/network-stack';
import { SecurityGroupStack } from '../lib/stack/security-group-stack';

const app = new cdk.App();
const networkStack = new NetworkStack(app, 'NetworkStack', {});
new SecurityGroupStack(app, 'SecurityGroupStack', networkStack.vpc);