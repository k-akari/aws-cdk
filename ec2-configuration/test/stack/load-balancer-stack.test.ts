import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as Network from '../../lib/stack/network-stack';
import * as Iam from '../../lib/stack/iam-stack';
import * as SecurityGroup from '../../lib/stack/security-group-stack';
import * as Server from '../../lib/stack/server-stack';
import * as LoadBalancer from '../../lib/stack/load-balancer-stack';

test('ServerStackSnapshot', () => {
  const serviceName = 'service';
  const envType = 'test';
  const app = new App({
    context: {
      'serviceName': serviceName,
      'envType': envType
    }
  });

  const networkStack = new Network.NetworkStack(app, 'NetworkStack');
  const iamStack = new Iam.IamStack(app, 'IamStack');
  const securityGroupStack = new SecurityGroup.SecurityGroupStack(app, 'SecurityGroupStack', networkStack.vpc);
  const serverStack = new Server.ServerStack(app, 'ServerStack', networkStack.subnet.public1a, iamStack.iamRole.instanceProfile, securityGroupStack.sg.ec2);
  const loadBalancerStack = new LoadBalancer.LoadBalancerStack(app, 'LoadBalancerStack', networkStack.vpc, networkStack.subnet, securityGroupStack.sg.ec2, serverStack.ec2.mainInstance);
  const template = Template.fromStack(loadBalancerStack);

  expect(template).toMatchSnapshot();
});
