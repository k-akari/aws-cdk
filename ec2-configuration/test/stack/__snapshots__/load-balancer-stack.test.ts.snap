// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "Alb": Object {
      "Properties": Object {
        "IpAddressType": "ipv4",
        "Name": "service-test-alb",
        "Scheme": "internet-facing",
        "SecurityGroups": Array [
          Object {
            "Fn::ImportValue": "SecurityGroupStack:ExportsOutputFnGetAttSecurityGroupEc2GroupId103D9309",
          },
        ],
        "Subnets": Array [
          Object {
            "Fn::ImportValue": "NetworkStack:ExportsOutputRefSubnetPrivate1aAFF32B55",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "AlbListener": Object {
      "Properties": Object {
        "DefaultActions": Array [
          Object {
            "ForwardConfig": Object {
              "TargetGroups": Array [
                Object {
                  "TargetGroupArn": Object {
                    "Ref": "AlbTargetGroup",
                  },
                  "Weight": 1,
                },
              ],
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "Alb",
        },
        "Port": 80,
        "Protocol": "HTTP",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "AlbTargetGroup": Object {
      "Properties": Object {
        "Name": "service-test-tg",
        "Port": 80,
        "Protocol": "HTTP",
        "TargetType": "instance",
        "Targets": Array [
          Object {
            "Id": Object {
              "Fn::ImportValue": "ServerStack:ExportsOutputRefEc2Instance1aAF71BD6F",
            },
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "NetworkStack:ExportsOutputRefVpc8378EB38272D6E3A",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
